controllers/extendeddaemonset/controller.go

```diff
@@ -215,23 +215,47 @@ func (r *Reconciler) updateInstanceWithCurrentRS(logger logr.Logger, daemonset *
 		newDaemonset.Status.State = datadoghqv1alpha1.ExtendedDaemonSetStatusStateRunning
 		newDaemonset.Status.IgnoredUnresponsiveNodes = current.Status.IgnoredUnresponsiveNodes
 	}
+	metaNow := metav1.NewTime(now)
 
 	var updateDaemonsetSpec bool
 	// If the deployment is in Canary phase, then update status (and spec as needed)
 	if daemonset.Spec.Strategy.Canary != nil {
+
+		isCanaryPaused, reason := IsCanaryDeploymentPaused(daemonset.GetAnnotations(), upToDate)
+		isCanaryFailed := IsCanaryDeploymentFailed(daemonset.GetAnnotations(), upToDate)
+		if isCanaryFailed {
+			msg := fmt.Sprintf("canary failed with ers: %s", upToDate.GetName())
+			conditions.UpdateExtendedDaemonSetStatusCondition(&newDaemonset.Status, metaNow, datadoghqv1alpha1.ConditionTypeEDSCanaryFailed, corev1.ConditionTrue, "CanaryFailed", msg, false, false)
+			newDaemonset.Status.State = datadoghqv1alpha1.ExtendedDaemonSetStatusStateCanaryFailed
+		} else {
+			conditions.UpdateExtendedDaemonSetStatusCondition(&newDaemonset.Status, metaNow, datadoghqv1alpha1.ConditionTypeEDSCanaryFailed, corev1.ConditionFalse, "", "", false, false)
+		}
+
+		if isCanaryPaused && !isCanaryFailed {
+			msg := fmt.Sprintf("canary paused with ers: %s", upToDate.GetName())
+			conditions.UpdateExtendedDaemonSetStatusCondition(&newDaemonset.Status, metaNow, datadoghqv1alpha1.ConditionTypeEDSCanaryPaused, corev1.ConditionTrue, string(reason), msg, false, false)
+			newDaemonset.Status.State = datadoghqv1alpha1.ExtendedDaemonSetStatusStateCanaryPaused
+			newDaemonset.Status.Reason = reason
+		} else {
+			conditions.UpdateExtendedDaemonSetStatusCondition(&newDaemonset.Status, metaNow, datadoghqv1alpha1.ConditionTypeEDSCanaryPaused, corev1.ConditionFalse, "", "", false, false)
+		}
+
 		switch {
-		case IsCanaryDeploymentFailed(daemonset.GetAnnotations()):
+		case isCanaryFailed:
 			// Canary deployment is no longer needed because it was marked as failed
 			// This `if` block needs to be first to respect Failed Canary state until annotation is removed
 			newDaemonset.Status.Canary = nil
 			newDaemonset.Status.State = datadoghqv1alpha1.ExtendedDaemonSetStatusStateCanaryFailed
+			newDaemonset.Status.Reason = ""
 			// Restore active replicaset template. Note: this requires a full daemonset update
 			newDaemonset.Spec.Template = current.Spec.Template
 			updateDaemonsetSpec = true
+			clearCanaryAnnotations(newDaemonset)
```

Comment:
i had another thought - what do you think of moving this function to the next `case` block? that way we can reset all canary annotations when a canary ends. this came up while adding additional functionality to the `unpause` feature and needed to reset an annotation that indicates a canary was manually unpaused.

Reply:
the problem is that we need to remove the annotations when we detect the canary failed. 
if we don't remove the annotations in the `case` we will always enter to this case.

But I'm agree with you, we can also call this `clearCanaryAnnotations ` in the next `case`.
I'm doing the change

Reply:
ah, right. OK thank you!